docs/source/developers/contributing.rst

```diff
@@ -17,53 +17,73 @@
+the mailing lists give participants the opportunity to slow down and be
+thoughtful in their responses, and they help developers who are spread across
+many timezones to participate more equally.
 
-Note that you must be subscribed to the mailing list in order to post to it. To
-subscribe, send a blank email to dev-subscribe@arrow.apache.org.
+See `the community page <https://arrow.apache.org/community/>`_ for links to
+subscribe to the mailing lists and to view archives.
 
-Mailing list archives can be found `here <https://lists.apache.org/list.html?dev@arrow.apache.org>`_.
+Report bugs and request features
+================================
 
-Issue Tracking
-==============
+Using the software and sharing your experience is a very helpful contribution
+itself. Those who actively develop Arrow need feedback from users on what
+works and what doesn't. Alerting us to unexpected behavior and missing features,
+even if you can't solve the problems yourself, help us understand and prioritize
+work to improve the libraries.
 
-We use the `ASF JIRA <https://issues.apache.org/jira/projects/ARROW/issues>`_
+We use `JIRA <https://issues.apache.org/jira/projects/ARROW/issues>`_
 to manage our development "todo" list and to maintain changelogs for releases.
 In addition, the project's `Confluence site <https://cwiki.apache.org/confluence/display/ARROW>`_
 has some useful higher-level views of the JIRA issues.
 
 To create a JIRA issue, you'll need to have an account on the ASF JIRA, which
 you can `sign yourself up for <https://issues.apache.org/jira/secure/Signup!default.jspa>`_.
-The JIRA server hosts bugs and issues for multiple Apache projects.  The JIRA
+The JIRA server hosts bugs and issues for multiple Apache projects. The JIRA
 project name for Arrow is "ARROW".
 
-Before you create a new bug entry, we recommend you first
-`search <https://issues.apache.org/jira/projects/ARROW/issues/ARROW-5140?filter=allopenissues>`_
+You don't need any special permissions on JIRA to be able to create issues.
+Once you are more involved in the project and want to do more on JIRA, such as
+assign yourself an issue, you will need "Contributor" permissions on the
+Apache Arrow JIRA. To get this role, ask on the mailing list for a project
+maintainer's help.
+
+Tips for using JIRA
++++++++++++++++++++
+
+Before you create a new issue, we recommend you first
+`search <https://issues.apache.org/jira/issues/?jql=project%20%3D%20ARROW%20AND%20resolution%20%3D%20Unresolved>`_
 among existing Arrow issues.
 
 When reporting a new issue, follow these conventions to help make sure the
 right people see it:
 
-* Enter the component your issue pertains to (for example "Python" or "C++").
+* Use the **Component** field to indicate the area of the project that your
+  issue pertains to (for example "Python" or "C++").
 * Also prefix the issue title with the component name in brackets, for example
   ``[Python] issue name`` ; this helps when navigating lists of open issues.
```

Comment:
May want to mention here that the component name helps generate more readable changelogs when we release